templates/purge-registry.py.j2

```diff
@@ -0,0 +1,142 @@
+        self.name           = name
+        self.owner          = owner
+        self.creation_date  = datetime.datetime.strptime(creation_date, DATE_FORMAT)
+        self.version        = version
+ 
+    def should_be_purge(self, retention_time):
+        if (self.creation_date + datetime.timedelta(days=retention_time)) < datetime.datetime.now() :
+            return True
+        else:
+            logging.info('The artifact %s is %s old' % (self.name, abs((datetime.datetime.now() - self.creation_date))))
+            return False
+
+def get_call(url, basic):
+    logging.debug('HTTP GET call to %s' % url)
+    res=requests.get(url, auth=basic)
+    res.raise_for_status()
+    return res
+
+def delete_call(url, basic):
+    logging.debug('HTTP DELETE call to %s' % url)
+    res=requests.delete_call(url,auth=basic)
+    res.raise_for_status()
+    return res
+
+def get_organization(url, basic):
+    logging.info('Querrying all the organization')
+    try:
+        orgs=[]
+        res=get_call('%s/api/v1/orgs' % url, basic)
+        for elt in res.json():
+            orgs.append(elt['username'])
+        return orgs
+    except requests.exceptions.HTTPError as err:
+        logging.error('Error %s when trying to get the organization: %s' % (err.response.status_code, err.response.text))
+
+def get_artifacts(url, basic, org):
+    logging.info('Querrying org %s to list all artifacts' % org)
+    try:
+        arts=[]
+        res=get_call('%s/api/v1/packages/%s' % (url, org), basic)
+        for elt in res.json():
+            if elt['type'] == 'container':
+                arts.append(Artifact(elt['name'], elt['owner'], elt['created_at'], elt['version']))
+        return arts
+    except requests.exceptions.HTTPError as err:
+        logging.error('Error %s when trying to get all the artifact of %s: %s' % (err.response.status_code, org, err.response.text))
+        raise err
+
+def delete_artifact(url, basic, art):
+    logging.info('Deleting artifact %s of %s' % (art.name, art.owner))
+    try:
+        res=delete_call('%s/api/v1/packages/%s/container/%s/%s' % (url, art.owner, art.name, art.version), basic)
+    except request.exceptions.HTTPError as err:
+        logging.error('Error %s when trying to delete artifact %s:%s from %s: %s' % (err.response.status_code, art.name, art.version, art.owner, err.response.text))
+        raise err
+
+def prepare_credentials(args):
+    username=None
+    password=None
+    if args.username is not None:
+        logging.info('Loading username from parameter')
+    if args.password is not None:
+        logging.warn('Loading password from parameter, not recommended')
+    if environ.get('GITEA_USERNAME') is not None:
+        logging.info('Loading username from environment variable')
+        username=environ.get('GITEA_USERNAME')
+    if environ.get('GITEA_USERNAME') is not None:
+        logging.info('Loading password from environment variable')
+        username=environ.get('GITEA_PASSWORD')
```

Comment:
Same as in the other init script. Just use `default` and `assert` in `parse_args()`.